doc/man3/SSL_CTX_set_tlsext_servername_callback.pod

```diff
@@ -123,22 +123,29 @@ client is processed. The servername, certificate and ALPN callbacks occur after
 a servername extension from the client is processed.
 
 SSL_get_servername_type() returns the servername type or -1 if no servername
-is present. Currently the only supported type (defined in RFC3546) is
-B<TLSEXT_NAMETYPE_host_name>.
+is present.
+The only type defined in RFC 3546 and supported here is B<TLSEXT_NAMETYPE_host_name>.
 
-SSL_set_tlsext_host_name() sets the server name indication ClientHello extension
-to contain the value B<name>. The type of server name indication extension is set
-to B<TLSEXT_NAMETYPE_host_name> (defined in RFC3546).
+SSL_set_tlsext_host_name() sets the Server Name Indication (SNI)
+ClientHello extension defined in RFC 3546 section 3.1
+to contain the value I<name> of type B<TLSEXT_NAMETYPE_host_name>.
+If I<name> is NULL it clears the SNI extension.
+
+Using this function may be crucial also for correct routing of connection requests.
+TLS clients are recommended to use this function
+as well as L<X509_VERIFY_PARAM_set1_host(3)> or L<X509_VERIFY_PARAM_add1_host(3)>
```

Comment:
I would instead recommend the use of `SSL_set1_host` which is more straightforward to use (you don't have to first get hold of the `X509_VERIFY_PARAM` and handles IP addresses more correctly)

Reply:
Good point - done.
So far I was not aware of `SSL_{set1,add1}_host()`.